source/common/grpc/google_async_client_impl.cc

```diff
@@ -72,10 +111,12 @@ GoogleAsyncStreamImpl::GoogleAsyncStreamImpl(GoogleAsyncClientImpl& parent,
                                              const Protobuf::MethodDescriptor& service_method,
                                              AsyncStreamCallbacks& callbacks,
                                              const Optional<std::chrono::milliseconds>& timeout)
-    : parent_(parent), service_method_(service_method), callbacks_(callbacks), timeout_(timeout),
-      completion_thread_(new Thread::Thread([this] { completionThread(); })) {}
+    : parent_(parent), tls_(parent_.tls_), dispatcher_(parent_.dispatcher_), stub_(parent_.stub_),
+      service_method_(service_method), callbacks_(callbacks), timeout_(timeout) {}
 
-GoogleAsyncStreamImpl::~GoogleAsyncStreamImpl() { ASSERT(rw_ == nullptr); }
+GoogleAsyncStreamImpl::~GoogleAsyncStreamImpl() {
+  ENVOY_LOG(debug, "GoogleAsyncStreamImpl destruct");
```

Comment:
nit: not sure if it's worth leaving this debug print here or not. Up to you.

Reply:
It's been fairly useful when chasing down races, on balance I think it's better to leave it.